vcd/resource_vcd_vapp_vm.go

```diff
@@ -238,6 +210,82 @@ func resourceVcdVAppVmCreate(d *schema.ResourceData, meta interface{}) error {
 	return resourceVcdVAppVmUpdate(d, meta)
 }
 
+// Adds existing org VDC network to VM network configuration
+// Returns configured OrgVDCNetwork for Vm, networkName, error if any occur
+func addVdcNetwork(d *schema.ResourceData, vdc govcd.Vdc, vapp govcd.VApp, vcdClient *VCDClient) (vdcNetworks []*types.OrgVDCNetwork, networkName string, err error) {
+
+	netName := ""
+	net, err := vdc.FindVDCNetwork(d.Get("network_name").(string))
+
+	if err == nil {
+		netName = net.OrgVDCNetwork.Name
+	}
+
+	vdcNetworks = []*types.OrgVDCNetwork{net.OrgVDCNetwork}
+
+	vAppNetworkConfig, err := vapp.GetNetworkConfig()
+
+	vAppNetworkName := ""
+	if vAppNetworkConfig.NetworkConfig != nil {
+		vAppNetworkName = vAppNetworkConfig.NetworkConfig[0].NetworkName
+		if netName == "" {
+			net, err = vdc.FindVDCNetwork(vAppNetworkName)
+			if err != nil {
+				return []*types.OrgVDCNetwork{}, "", fmt.Errorf("error finding vApp network: %#v", err)
+			}
+
+			netName = net.OrgVDCNetwork.Name
+		}
+
+	} else {
+
+		if netName == "" {
+			return []*types.OrgVDCNetwork{}, "", fmt.Errorf("'network_name' must be valid when adding VM to raw vApp")
+		}
+
+		err = retryCall(vcdClient.MaxRetryTimeout, func() *resource.RetryError {
+			task, err := vapp.AddRAWNetworkConfig(vdcNetworks)
+			if err != nil {
+				return resource.RetryableError(fmt.Errorf("error assigning network to vApp: %#v", err))
+			}
+			return resource.RetryableError(task.WaitTaskCompletion())
+		})
+
+		if err != nil {
+			return []*types.OrgVDCNetwork{}, "", fmt.Errorf("error assigning network to vApp:: %#v", err)
+		} else {
+			vAppNetworkName = netName
+		}
+
+	}
+
+	if vAppNetworkName != netName {
+		return []*types.OrgVDCNetwork{}, "", fmt.Errorf("the VDC network '%s' must be assigned to the vApp. Currently the vApp network data is %s", netName, vAppNetworkName)
+	}
+
+	log.Printf("[TRACE] Network name found: %s", netName)
+
+	return vdcNetworks, netName, nil
+}
+
+// Adds existing org vApp network to VM network configuration
```

Comment:
Same here I don't agree with having this function added to this code base it should be in underlying go-vcloud-director library.

Reply:
This functionality in latest commits was changed and just works as check.